### PR TITLE
Node-aware Context Menu Items

### PR DESCRIPTION
**Pull request by ivailop7**
Expose a 'onWillOpen' method that passes the 'contextmenu' mouse event, so users can do their logic what items should end up in the context menu before it's rendered. Have added an example for 'Link node', that appends an item to 'Remove Link' when a link node is right-clicked. Also fixed wrong positioning of the context menu when the page is scrolled.

### CODE REVIEW

File: packages/lexical-playground/src/plugins/ContextMenuPlugin/index.tsx [hunk 1]
```diff
@@ -199,10 +203,32 @@ export default function ContextMenuPlugin(): JSX.Element {
     [editor],
   );
 
+  const onWillOpen = (event: MouseEvent) => {
+    let newOptions = defaultOptions;
+    editor.update(() => {
```
Comment (acywatson) [inline packages/lexical-playground/src/plugins/ContextMenuPlugin/index.tsx:208]: This should probably be a read that returns the node type, then call an update onClick

File: packages/lexical-react/src/LexicalContextMenuPlugin.tsx [hunk 2]
```diff
@@ -97,6 +99,9 @@ export function LexicalContextMenuPlugin<TOption extends MenuOption>({
   const handleContextMenu = useCallback(
     (event: MouseEvent) => {
       event.preventDefault();
+      if (onWillOpen) {
```
Comment (acywatson) [inline packages/lexical-react/src/LexicalContextMenuPlugin.tsx:102]: Can we be strict with the check here, like for onOpen and onClose?

File: packages/lexical-playground/src/plugins/ContextMenuPlugin/index.tsx (hunk 1 above)
Comment (ivailop7) [reply packages/lexical-playground/src/plugins/ContextMenuPlugin/index.tsx:208]: $getNearestNodeFromDOMNode works only inside editor.update(), not unsure on why this is the case.

File: packages/lexical-react/src/LexicalContextMenuPlugin.tsx (hunk 2 above)
Comment (ivailop7) [reply packages/lexical-react/src/LexicalContextMenuPlugin.tsx]: updated

**Review by acywatson — APPROVED** [review]

**Pull request merged**